src/material/schematics/ng-update/upgrade-rules/hammer-gestures-v9/hammer-gestures-rule.ts

```diff
@@ -134,7 +134,51 @@ export class HammerGesturesRule extends MigrationRule<null> {
     const hasCustomGestureConfigSetup =
         this._hammerConfigTokenReferences.some(r => this._checkForCustomGestureConfigSetup(r));
 
-    if (this._usedInRuntime || this._usedInTemplate) {
+    /*
+      Possible scenarios and how the migration should change the project:
+        1. We detect that a custom HammerJS gesture config set up:
+            - Remove references to the Material gesture config if no event from the
+              Angular Material gesture config is used.
+            - Print a warning about ambiguous setup that cannot be handled completely.
+        2. We detect that HammerJS is *only* used at runtime:
```

Comment:
I'm not 100% clear on what this means. Only used by Angular Material (previously)? or only used by their app? This is programmatic access through JS APIs?

Reply:
Will try to make that more clear.